simln-lib/src/sim_node.rs

```diff
@@ -1211,6 +1216,77 @@ impl SimNetwork for SimGraph {
     }
 }
 
+/// This trait defines the "interface" for payment propagation logic
+pub trait PaymentPropagator: Send + Sync + 'static {
```

Comment:
This seems like a lot of boilerplate just for tests (and possibly somewhere we should use `async_trait` rather than have these boxes + pins?

Doesn't seem like it would be too much test code to just use real propagation and create payments that fail at the points we want?